definitions/features/mongo.rb

```diff
@@ -0,0 +1,124 @@
+class Features::Mongo < ForemanMaintain::Feature
+  # assume mongo is installed when there is Pulp
+  PULP_DB_CONFIG = '/etc/pulp/server.conf'.freeze
+
+  attr_reader :configuration
+  metadata do
+    label :mongo
+
+    confine do
+      feature(:pulp)
+    end
+  end
+
+  def data_dir
+    '/var/lib/mongodb'
+  end
+
+  def config_files
+    [config_file]
+  end
+
+  def config_file
+      PULP_DB_CONFIG
+  end
+
+  def initialize
+     @configuration = load_db_config(config_file)
+  end
+
+  def local?
+    ['localhost', '127.0.0.1', hostname].include?(configuration['host'])
+  end
+
+  def base_command(command, config = configuration, args = '')
+    if config['ssl']
+      ssl = " --ssl"
+      if config['ca_path']
+        ca_cert = " --sslCAFile #{config['ca_path']}"
+        client_cert = " --sslPEMKeyFile #{config['ssl_certfile']}" if config['ssl_certfile']
+      end
+    end
+    username = " -u #{config['username']}" if config['username']
+    password = " -p #{config['password']}" if config['password']
+    host = "--host #{config['host']} --port #{config['port']}"
+    cmd = "#{command}#{username}#{password} #{host}#{ssl}#{ca_cert}#{client_cert} #{args}"
+  end
+
+  def mongo_command(args, config = configuration)
+    base_command('mongo', config, "#{args} #{config['name']}")
+  end
+
+  def dump(target, config = configuration)
+    execute!(base_command('mongodump', config, "-d #{config['name']} --out #{target}"))
```

Comment:
whenever we use the base_command, we should also pass `hidden_patterns` option, similar as in https://github.com/theforeman/foreman_maintain/blob/49084935e82260f9c2c7aaba5618cd90af5a55a4/definitions/features/candlepin_database.rb#L34, to make sure we hide the sensitive information from logs.